### PR TITLE
Support signing SPRs and validating signatures.

### PR DESCRIPTION
**Pull request by aktary**
1) Minor: needed to update the version of factom we were depending on.  As a result, there's a minor change to `factom.GetTransaction()` in burns.go.
2) Since we have existing SPRs we have to honor, we need to set an activation height for signing, thus the new Version and update to activation.go.
3) Used `factom.SignData()` to request that the wallet sign the SPR... this _should_ work, but is untested so far.  We do this signging request in `CreateSprEntry()` to support multiple addresses.
4) Validate signature in `ValidateS1()`

### CODE REVIEW

File: spr/spr.go [hunk 1]
```diff
@@ -236,11 +237,21 @@ func (spr *StakingPriceRecord) CreateSPREntry() (*factom.Entry, error) {
...
-	e.ExtIDs = [][]byte{{spr.Version}, rcd, spr.GetHash()}
+
 	e.Content, err = spr.SafeMarshal()
 	if err != nil {
 		return nil, err
 	}
+
+	signature, err := factom.SignData(spr.CoinbaseAddress, e.Content)
+	if err != nil {
+		return nil, err
```
Comment (dragonheaven) [inline spr/spr.go:248]: I see ` Method not found` error in this code line. @aktary

Comment (aktary) [reply spr/spr.go:248]: Resolved by using the latest alpine factom-walletd docker image.  docker-compose.yml updated.

File: balances/burns.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func (b *BurnTracking) UpdateBurns(c *config.Config, startBlock int64) error {
 		}
 
 		for _, txid := range fc.Transactions {
-			txInterface, err := factom.GetTransaction(txid.TxID)
+			txInterface, err := factom.GetTransaction(string(txid))
```
Comment (dragonheaven) [inline balances/burns.go:51]: Is there any special reason you changed the txid format here?

File: modules/graderStake/s1_util.go [hunk 3]
```diff
@@ -40,6 +42,13 @@ func ValidateS1(entryhash []byte, extids [][]byte, height int32, content []byte)
 	}
 	o := &spr.S1Content{V2Content: *o2}
 
+	if exids[0][0] == 6 {
```
Comment (dragonheaven) [inline modules/graderStake/s1_util.go:45]: variable is not correct. it should be `extids ` instead of `exids`.  @aktary

Comment (aktary) [reply modules/graderStake/s1_util.go:45]: 😵

**Comment by StarNeit** [comment]
This has been merged into master already, thanks

**Pull request closed without merging**